### DISCUSSION TITLE
How to Stop Closing Application when Ctrl+Q is Executed?

### DISCUSSION THREAD

**Discussion by ichan-akira** [tui-cs/Terminal.Gui#1413] [Q&A] [answered]
Pretty much the same question as #633 but now for Ctrl+Q:

How can I prevent the console application from closing when `Ctrl+Q` is executed?

**Comment by BDisp** [comment, marked as answer]
Try this:

```cs
Application.Top.KeyPress += (e) => {
	if (e.KeyEvent.Key == (Key.CtrlMask | Key.Q)) {
		// Do nothing and return true
		e.Handled = true;
	}
};

```

**Reply by ichan-akira** [reply]
Indeed, this is the solution that I seek for! 👍 

I just hope that this is actually documented somewhere (or I definitely missed it).

**Reply by ichan-akira** [reply]
out of topic: This discussion layout remind me of stackoverflow! nice github

**Reply by BDisp** [reply]
I really do not know if this is currently documented, probably not, but it's for this that this `KeyPress` event is used.

**Comment by BDisp** [comment]
> Pretty much the same question as #633 but now for Ctrl+Q:

Is a different question. On old versions and on all OS, pressing Ctrl+C forces the application to close. Then I added flags to respective drivers to avoiding that.

**Reply by ichan-akira** [reply]
Ah I see, yes indeed in that sense, this is indeed different question.
